src/main/java/edu/usf/cutr/gtfsrtvalidator/background/BackgroundTask.java

```diff
@@ -199,16 +199,15 @@ public void run() {
             feedMessageBuilder.addAllEntity(allEntitiesArrayList);
 
             GtfsRealtime.FeedMessage combinedFeed = feedMessageBuilder.build();
-
-            // Use the same current time for all rules for consistency
-            long currentTimeMillis = System.currentTimeMillis();
-
-            // Run validation rules
-            for (FeedEntityValidator rule : mValidationRules) {
-                validateEntity(currentTimeMillis, combinedFeed, previousFeedMessage, gtfsData, gtfsMetadata, feedIteration, rule);
+            synchronized(_log) {
```

Comment:
So does the underlying problem here seem to be that the `org.slf4j.Logger` isn't thread-safe?

https://stackoverflow.com/a/18543651/937715 seems to say it is.

I'd like to avoid synchronizing the entire execution of the rules if possible, as we'll want this to run in parallel when possible (e.g., when processing archived files - https://github.com/CUTR-at-USF/gtfs-realtime-validator/issues/68).  As proposed here, we'd wait for one GTFS-rt feed to completely finish rule validation before the next could start.  This could block in the order of seconds for really big feeds.

Do you think synchronizing the `GtfsUtils.logDuration()` would have the same effect as the proposed change here?